### ISSUE TITLE
Output variable for release notes doesn't persist between phases

### ISSUE THREAD

**Issue by hintbw** [rfennell/vNextBuild#276]
### VSTS Extension you are using
*There are a number of VSTS extension in this repo so please say which one you are using, and if an extension contains multiple tasks which task also*

Generate Release Notes extension (Powershell based)

### Where are you running it?
*VSTS or on-premise TFS (if so what version)*

VSTS

### Version of Extension/Task
*I assume it is the currently public version, if you are using an older version on-premises let me know, but the answer will probably involve upgrading to the currently public version*

2.19.2

### Expected behaviour and actual behaviour

I am using an agent phase to generate the release notes and using the output variable (variable name releaseNotes) to pass back to the release pipleine so I can reuse that information elsewhere in the pipeline. The variable is not being evaluated in another task (that is in a separate phase) and it is just displayed as $(releaseNotes)

### Steps to reproduce the problem

Create an agent phase in a release pipeline that uses the Generate release notes task. Set an output variable of releaseNotes on that task so that it contains what is generated by the task.

Create an agentless phase and add a Manual Intevention step. In the instructions part of the manual intervention task add a reference to the $(releaseNotes) variable.

When the release pipeline is run, the $(releaseNotes) variable will not show the actual information that was stored in it, instead it will just display $(releaseNotes) in the manual intervention dialog box.

**Comment by rfennell** [comment]
I am not sure that variables are meant to be persisted between phases, being that phases can run different VMs.

I need to do some research

**Comment by rfennell** [comment]
I have checked with Microsoft and it seems that passing variables between phases is not currently supported

**Comment by hintbw** [comment]
@rfennell - Thanks for that info - I'm not surprised as I have seen many asking that question and no great answers about how someone has implemented that successfully.

In the September 2017 release of VSTS (Spring 123) they did mention they were working on the following items for multi-phase builds and releases:

We’re still working on a few features, including:

Ability to select a different queue in each phase.
**Ability to consume output variables from one phase in a subsequent phase.**
Ability to run phases in parallel. (For now, all the phases you define run sequentially).

This feature would be helpful for the release notes output variable in the future, but I understand it isn't now currently possible.

**Comment by rfennell** [comment]
I agree it is a feature we are waiting for.

In the case of this task are there any options by writing the text to a known file location all the agents can see?

**Comment by rfennell** [comment]
I have heard more from Microsoft...

If you define your release pipeline with YAML it is possible. It is documented at [YAML getting started with phases](https://github.com/Microsoft/vsts-agent/blob/master/docs/preview/yamlgettingstarted-phases.md  )

Currently the VSTS UI uses an older definition model and so cross phase variables are not available if you edit your releases graphically. 

Sorry don't now when the UI will catch up with the newer YAML way of working.

Hope that helps

**Comment by rfennell** [comment]
As this is beyond my immediate control UI wise I am closing this issue, re-open if needed
